WoWsShipBuilder.Core/DataContainers/AccelerationHelper.cs

```diff
@@ -0,0 +1,416 @@
+    /// <param name="maxPowerBackwards">The maximum power backwards.</param>
+    /// <returns>The power of the given throttle.</returns>
+    private static double GetPowerFromThrottle(double throttle, double maxPowerForward, double maxPowerBackwards)
+    {
+        if (throttle > 0)
+        {
+            return maxPowerForward * Math.Pow(throttle / 4, 2);
+        }
+        else
+        {
+            return maxPowerBackwards;
+        }
+    }
+
+    /// <summary>
+    /// Return if the calculation cycle should continue or stop.
+    /// </summary>
+    /// <param name="newThrottle">Current throttle to reach.</param>
+    /// <param name="oldThrottle">Starting throttle.</param>
+    /// <param name="speed">Current speed.</param>
+    /// <param name="speedLimit">Current Speed limit.</param>
+    /// <returns>If the cycle should continue.</returns>
+    private static bool ShouldCycle(double newThrottle, double oldThrottle, double speed, double speedLimit)
+    {
+        // we are accelerating/going towards max speed. Graph is going upwards
+        if (newThrottle > oldThrottle)
+        {
+            return speed < speedLimit - Margin;
+        }
+        else
+        {
+            // we are decelerating/going towards reverse speed. Graph is going downwards.
+            return speed > speedLimit + Margin;
+        }
+    }
+
+    /// <summary>
+    /// Calculate a single iteration for the acceleration.
+    /// </summary>
+    /// <param name="pointList">List to which to add the calculated point.</param>
+    /// <param name="throttle">Current throttle.</param>
+    /// <param name="isReversingDirection">If the ship is reversing direction from forward to backward.</param>
+    /// <param name="time">Current time, passed as ref. Will get modified.</param>
+    /// <param name="speed">Current speed, passed as ref. Will get modified.</param>
+    /// <param name="speedLimit">Current speed limit.</param>
+    /// <param name="power">Current power, passed as ref. Will get modified.</param>
+    /// <param name="powerIncreaseForward">Increase of power when going forward.</param>
+    /// <param name="maxPowerForward">Maximum power forward.</param>
+    /// <param name="maxForwardSpeed">Maximum speed forward.</param>
+    /// <param name="forsageForwardMaxSpeed">Maximum forsage speed forward.</param>
+    /// <param name="forsageForward">Forsage forward.</param>
+    /// <param name="powerIncreaseBackward">Increase of power when going backwards.</param>
+    /// <param name="maxPowerBackwards">Maximum power backwards.</param>
+    /// <param name="maxReverseSpeed">Maximum speed backwards.</param>
+    /// <param name="forsageBackwardsMaxSpeed">Maximum forsage speed backwards.</param>
+    /// <param name="forsageBackwards">Forsage backwards.</param>
+    /// <param name="stoppingFromBackward">If the ship is stopping from backward.</param>
+    private static void GenerateAccelerationPoints(
```

Comment:
As before, refactor to reduce number of parameters

Reply:
This would require a record just for this private method. it would just introduce one layer of encapsulation for pretty much no reason. It really makes no sense to inrduce a record for a one time usage